src/Domain/Donation.php

```diff
@@ -63,9 +63,9 @@ class Donation extends SalesforceWriteProxy
 
     /**
      * @ORM\Column(type="datetime", nullable=true)
-     * @var ?DateTime    When the donation first moved to status Collected, i.e. the donor finished paying.
+     * @var ?DateTimeInterface  When the donation first moved to status Collected, i.e. the donor finished paying.
      */
-    protected ?DateTime $collectedAt = null;
+    protected ?DateTimeInterface $collectedAt = null;
```

Comment:
If we're moving away from DateTime could we type it as `DateTimeImmutable`? I think the existence of that interface is a mistake, (and the author of it agrees) since the two classes are not really interchangable.

Would require changing the ORM mapping to `type="datetime_immutable"` but shouldn't need any DB migration.

Reply:
I think this does need a migration to keep Doctrine happy because it stores a comment about the type to ensure it's handled consistently. That seems fine though – it's cool that this has a native Doctrine type now!